src/main/java/uk/gov/ons/ctp/response/sample/message/SampleOutboundPublisher.java

```diff
@@ -0,0 +1,12 @@
+package uk.gov.ons.ctp.response.sample.message;
+
+import uk.gov.ons.ctp.common.error.CTPException;
+import uk.gov.ons.ctp.response.sample.domain.model.SampleSummary;
+
+/**
+ * An interface for publishing notifications that a sample has been uploaded
+ */
+public interface SampleOutboundPublisher {
+    void sampleUploadStarted(SampleSummary sampleSummary) throws CTPException;
```

Comment:
Why are we posting a message when the process of uploading starts?

Reply:
If you think about the sample service in isolation, uncoupled from other services, how can you know what properties of the sample service are interesting?  If you are providing a feature which publishes sample upload status, it seemed logical to include both the start and the end of the process.

Reply:
I don't understand why we'd implement code that might never be used. If there was a business requirement to know when the sample has started being uploaded I wouldn't question it. Adding stuff because we "might" need it isn't a good justification in my opinion. Code left around that isn't used costs time when people will try to understand "oh there is a message being published when this is started, I wonder what consumes it"

Reply:
It is important to make well rounded APIs.  Each service is it's own product and needs to be seen as such.  Our motley collection of half finished APIs does not nearly achieve this and makes it difficult to test.  For example, no update or delete endpoints make it hard to do full lifecycle testing.  IMHO those endpoints should be there even if there is no current concrete business requirement for it.  So where do you draw the line?  The case for this message is weaker.  I take the view that this feature requires the lifecycle of an upload to be published (which in business terms is what it is).  As such the start message has as much business being there as the finished one